### PR TITLE
[codegen/docs] Only emit first 200 nested types

### PR DESCRIPTION
**Pull request by lukehoban**
There are a few resources with a *very* large number of nested types.  We have seen this lead to generating docs pages that cannot be loaded in a browser.

Instead, only emit the first 200 nested types, and emit a note to users if we have done this.

In the AWS provider, this only triggers for `wafv2.WebAcl` and `wafv2.RuleGroup`.

This reduces the size of http://localhost:1313/registry/packages/aws/api-docs/wafv2/webacl/ from 64MB to 2.5MB (still huge - but manageable).

Screenshot 2023-02-03 at 3 28 40 PM

Fixes https://github.com/pulumi/registry/issues/1481 (once this is incorporate into `registrygen`).

### CODE REVIEW

**Review by abhinav — APPROVED** [review]

File: pkg/codegen/docs/templates/resource.tmpl [hunk 1]
```diff
@@ -296,14 +296,20 @@ The following state arguments are supported:
 
 ## Supporting Types
 
-{{ range .NestedTypes }}
+{{ if ge (len .NestedTypes) 200 }}
+> **Note:** There are over 200 nested types for this resource.
+Only the first 200 types are included in this documentation.
```
Comment (susanev) [inline pkg/codegen/docs/templates/resource.tmpl:301]: is it possible to use the hugo shortcode for our note styles here?

also, can we tell them what we recommend that they do to access the additional types?

**Review by justinvp — APPROVED** [review]
Nice

**Review by susanev — APPROVED** [review]
thank you for fixing this!! 🎉

**Comment by justinvp** [comment]
Looks like the change to the template is introducing some minor whitespace changes that don't match expected output. You'll need to re-run the [`TestGeneratePackage`](https://github.com/pulumi/pulumi/blob/7d53b479ed387acd5d45e583e90e61737ba44712/pkg/codegen/docs/gen_test.go#L516) with `PULUMI_ACCEPT=true` and commit the changes.

**Comment by lukehoban** [comment]
> is it possible to use the hugo shortcode for our note styles here?

Honestly not sure.  I don't think we use hugo shortcodes in the existing docgen, but I could be wrong.  What we want this to be instead?  

> also, can we tell them what we recommend that they do to access the additional types?

Unfortunately due to the same underlying issues, anything we might suggest as an alternative will also have problems.  We could say something like "The types are present in your language SDK, and documentation for them is available via your IDE tooling".  But I fear that will also be frustrating.  I'm inclined to just leave this a little more open ended.

**Review by cnunciato — APPROVED** [review]
Looks good! 👍  

> is it possible to use the hugo shortcode for our note styles here?

It's possible, but we removed all Hugo shortcodes from these templates in https://github.com/pulumi/pulumi/pull/9491 and https://github.com/pulumi/pulumi/commit/af8e010a4a6fb609dd610d96b19bba73a136cfbc last year for performance reasons (they played a big part in the build failures we started seeing ahead of May 4), so while using one here probably wouldn't make much of a practical difference (assuming most resources have fewer than 200 subtypes), in general I'd recommend keeping to a no-shortcode policy for these.

**Comment by susanev** [comment]
@cnunciato okay sounds good, lets watch for continued questions about this to see if we should reconsider the prominence of the note.

**Comment by lukehoban** [comment]
bors merge

**Pull request merged**